### ISSUE TITLE
[Website] Improve website of the master version to prepare for the 1.0 release

### ISSUE THREAD

**Issue by sxjscience** [dmlc/gluon-nlp#1472]
## Description

https://github.com/dmlc/gluon-nlp/pull/1374 has been merged so we have fixed the warnings in our documents. However, the current structure of the website is not very satisfactory and we should try to improve the layout and also add more tutorials.

Help needed here.

## References
- Previous issue on website: https://github.com/dmlc/gluon-nlp/issues/1081

@dmlc/gluon-nlp-committers 
@Cli212 @yongyi-wu @xinyual

**Comment by sxjscience** [comment]
Some items:

- [ ] Improve the front-page demo
- [ ] Improve layout of the examples

**Comment by szha** [comment]
On the [Text Prediction - Part 1: Quickstart of Pretrained Backbones](https://nlp.gluon.ai/master/tutorials/text_prediction/text_prediction_part1.html)
- [ ] the start of the tutorial mentions `we will load two datasets using the nlp_data command...` but the immediate next block are imports. Consider mentioning that the first block is import.
- [ ] should there be a tutorial for nlp_data/nlp_process CLI?
- [ ] formatting issue `Let’s download two datasets from the GLUE benchmark: - The Standford Sentiment Treebank (SST-2) - Semantic Textual Similarity Benchmark (STS-B)`
- [ ] formatting issue `A bunch of recent papers, especially BERT, have led a new trend for solving NLP problems: - Pretrain a backbone model on a large corpus, - Finetune the backbone to solve the specific NLP task.`
- [ ] cell 13, the matrix values don't add to the content of the tutorial.
- [ ] cell 14, the list of backbones don't seem to provide much information. we could provide some comparison such as number of parameters, pre-training data domain, downstream task accuracy, etc., to help guide the model selection.
- [ ] Conclude sentence `According to the paper and our own experiments, MobileBERT performs similar to BERT-base...`, while useful as an introduction to the next tutorial, diverges from the topic of quick start. Consider concluding with something like "In this tutorial, we learned how to use pre-trained backbone networks to quickly start ... Next, we will ..."

**Comment by szha** [comment]
- [ ] I saw several notebooks with a lot of imports. Let's make sure we only include the necessary ones. The import order should be python built-ins, third-party libraries, and then first-party libraries.

**Comment by szha** [comment]
On the [Question Answering with GluonNLP](https://nlp.gluon.ai/master/tutorials/question_answering/question_answering.html)
- [ ] The mentions of `https://github.com/dmlc/gluon-nlp/tree/master/scripts/question_answering` should link to the example page so that we always refer to the source code in the same version.
- [ ] We should mention/provide pointer on how users can obtain their own checkpoints for `!wget -O google_electra_base_squad2.0_8160.params https://gluon-nlp-log.s3.amazonaws.com/squad_training_log/fintune_google_electra_base_squad_2.0/google_electra_base_squad2.0_8160.params`

**Comment by szha** [comment]
On the [Tokenization - Part1: Basic Usage of Tokenizer and Vocabulary](https://nlp.gluon.ai/master/tutorials/tokenization/tokenization_part1.html)
- [ ] `text processing workflow: raw text => normalized (cleaned) text => tokens => network` can be replaced with a diagram in which examples of each step can be shown.

**Comment by szha** [comment]
On the [Tokenization - Part2: Learn Subword Models with GluonNLP](https://nlp.gluon.ai/master/tutorials/tokenization/tokenization_part2.html)
- [ ] We should have an introductory section for the concepts of subwords before diving in to the implementation/execution

**Comment by szha** [comment]
On [Compile NLP Models - Convert GluonNLP Models to TVM](https://nlp.gluon.ai/master/tutorials/deep_learning_compiler/tvm_basic.html)

- [ ] At the moment this notebook includes only code and titles. We will need an introduction section for overview and motivation, explanation for the code, reference to other materials on compilation
- [ ] `compile_tvm_graph_runtime` is very long and there are opportunities for breaking up the logic into smaller functions. For example, graph construction can happen separately from compilation.
